welltestpy/tools/diagnostic_plots.py

```diff
@@ -58,15 +66,40 @@ def diagnostic_plot_pump_test(observation, fig=None, ax=None,plotname=None,style
         fig, ax = plotter._get_fig_ax(fig, ax)
         ax.scatter(x, y, color="red", label="drawdown")
         ax.plot(sx, sy, c="red")
-        ax.plot(dx, dy, c="black", linestyle='dashed')
-        ax.scatter(dx, dy, c="black", marker="+",  label="time derivative")
-        ax.set_xscale("symlog")
-        ax.set_yscale("symlog")
-        ax.set_ylim([0, head[-1]])
+        ax.plot(dx, dy, c="black", linestyle="dashed")
+        ax.scatter(dx, dy, c="black", marker="+", label="time derivative")
+        ax.set_xscale("symlog", linthresh=linthresh_time)
+        ax.set_yscale("symlog", linthresh=linthresh_head, base=10)
         ax.set_xlim(time[0], time[-1])
-        ax.set_xlabel("$t$ in [s]", fontsize=12)
-        ax.set_ylabel("$h$ and $dh/dx$ in [m]", fontsize=12)
-        ax.set_yticks(np.arange(0, head[-1], 0.02))
+        yticks = []
+        for i in [
+            -1e5,
+            -1e4,
+            -1e3,
+            -1e2,
+            -1e1,
+            -1e0,
+            -1e-1,
+            -1e-2,
+            1e-2,
+            1e-1,
+            1e0,
+            1e1,
+            1e2,
+            1e3,
+            1e4,
+            1e5,
+        ]:
+            if i > min(head[0], derivative[0]):
+                if i < max(head[-1], derivative[-1]):
+                    yticks.append(i)
+                else:
+                    continue
+            else:
+                continue
+        ax.set_yticks(yticks)
```

Comment:
https://raw.githubusercontent.com/GeoStat-Examples/welltestpy-field-site-analysis/master/results/03_unconfined/horkheim_p05p44_analysis.png
Here this is only done for `head` (ignoring the range of the `derivative`).

Reply:
And I guess we also need to use `np.abs(head)` to cut out negative values.

Reply:
Returns "OverflowError: cannot convert float infinity to integer" min_e will try to get the log10 of 0.

Reply:
Ok, then we need to cut this off:
```python
max_e = int(np.ceil(np.log10(max_v)))
if min_v < linthresh_head:
    ax.set_ylim(0, 10 ** max_e)
else:
    min_e = int(np.floor(np.log10(min_v)))
    ax.set_ylim(10 ** min_e, 10 ** max_e)
```

Reply:
It works. But when the head values are small the plotted lines remain in the bottom part of the graph so that trends are not well shown.

Reply:
This is then controllable with `linthresh_head` or am I wrong?

Reply:
I can't seem to change the plot when changing the linthresh_head. I think it has something to do with the np.ceil which creates, for the smallest values,  always a max of 10^0. This results in the too large y scale.

Reply:
The 01_create has also a negative pumping rate which messes with our y scale because we take the log10 of negative values. (and taking the abs(head) results in making positive  a y axis while working with a plot on the negative y axis)